docs/tutorial/flow-nft-marketplace.md

```diff
@@ -0,0 +1,1804 @@
+    let ownerAddress = PetStore.owners[id]!
+
+    // We encounter the `getAccount(_ addr: Address)` function again.
+    // Get the `AuthAccount` instance of the current owner.
+    let ownerAcct = getAccount(ownerAddress)
+
+    // Borrow the `NFTReceiver` capability of the owner.
+    let receiverRef = ownerAcct.getCapability<&{PetStore.NFTReceiver}>(/public/NFTReceiver)
+        .borrow()
+            ?? panic("Could not borrow receiver reference")
+
+    // Happily delegate this query to the owning collection
+    // to do the grunt work of getting its token's metadata.
+    return receiverRef.getTokenMetadata(id: id)
+}
+
+```
+
+Now, execute the script:
+
+```shell
+
+$ flow scripts execute src/flow/script/GetTokenMetadata.cdc <TOKEN_ID>
+
+```
+
+If we have minted an NFT with the metadata `{"name": "Max", "breed": "Bulldog"}` in the [previous minting step](#minttoken-transaction), then that is what you will get after running the script.
+
+### `GetAllTokenIds` (Bonus)
+
+This script is very short and straightforward, and it will become handy
+when we build a UI to query tokens based on their IDs.
+
+```cadence
+
+// GetAllTokenIds.cdc
+
+import PetStore from 0xPetStore
+
+pub fun main() : [UInt64] {
+    // We basically just return all the UInt64 keys of `owners`
+    // dictionary as an array to get all IDs of all tokens in existence.
+    return PetStore.owners.keys
+}
+
+```
+
+Et voila! You have come very far and dare I say you are ready to start building your own Flow NFT app.
+
+However, user experience is a crucial part in any app. It is more than likely that your users won't be as proficient at the command line as you do. Moreover, it is a bit boring for an NFT store to have faceless NFTs. In the next section, we will start tackling the fun part--building the UI on top and using [nft.storage][nft-storage] service to upload and store images of our NFTs.
+
+### Front-end app
+
+> **💡 Do you know React?**    
+> In this part of the tutorial, we will work with React.js extensively.
+> Because learning React is unfortunately out of the scope, if you think
+> you will need some introduction or brush-up on React, please head over
+> to [Intro to React][react-intro] tutorial before returning here.
+
+Our NFT pet store is fully functional at the command line, but without a face, it is too hard for end users to use.
+
+Very often, especially for [decentralized applications][dapps] whose back-ends rely heavily on blockchains and other decentralized technology, the user experience is what makes or breaks them. Quite often, the user-facing part *is* the only crucial part in a dapp.
+
+In this section, we will be working on the UI for the pet store app in React.js. While you're expected to have some familiarity with the library, I will do my best to use common features instead of trotting into advanced ones.
+
+After we are done, we will end up with a simple marketplace app that users can mint and query their NFTs that is similar to this:
```

Comment:
Which sentence in particular?

Reply:
I might split it into two sentences:

```suggestion
After we are done, we will end up with a simple marketplace app that users can mint and query their NFTs. It will be similar to this:
```